samples/insert-query-with-get-trasanction.js

```diff
@@ -0,0 +1,75 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// sample-metadata:
+//  title: Perform a dml insert query using getTransaction method with optimisticLock
+//  usage: node insert-query-with-get-transaction.js <INSTANCE_ID> <DATABASE_ID> <PROJECT_ID>
+
+async function main(instanceId, databaseId, projectId) {
+  // [START spanner_insert_query_with_get_transaction]
+  // Imports the Google Cloud client library.
+  const {Spanner} = require('@google-cloud/spanner');
+
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // const projectId = 'my-project-id';
+  // const instanceId = 'my-instance';
+  // const databaseId = 'my-database';
+
+  // Creates a client
+  const spanner = new Spanner({
+    projectId: projectId,
+  });
+
+  async function writeTransaction() {
+    // Gets a reference to a Cloud Spanner instance and database
+    const instance = spanner.instance(instanceId);
+    const database = instance.database(databaseId);
+
+    const options = {
+      optimisticLock: true,
+    };
+
+    const promise = await database.getTransaction(options).then();
```

Comment:
The use of `getTransaction(..)` means that there is no automatic retrying of the transaction if it is aborted by Spanner. Transaction aborts can happen at any time for any read/write transaction (not only because of lock conflicts, but also for more random, transient problems). That again means that this sample function will fail at random for a (very small) percentage of the times that it is called. Would it not make more sense to make this sample in such a way that it shows that the user is responsible for retrying transactions when they use `getTransaction(..)`?

Reply:
I have removed the sample, since we are recommending clients to use the runTransaction method also adding samples to use getTransaction method will need discussions with the documentation team for the tags